docs/tutorials/notes-mvvm/includes/viewmodel-notes.md

```diff
@@ -0,0 +1,119 @@
+
+## Notes viewmodel
+
+Based on what the **AllNotes view** is going to display and what interactions the user will do, the **Notes viewmodel** must provide the following items:
+
+- A collection of notes.
+- A command to handle navigating to a note.
+- A command to create a new note.
+- Update the list of notes when one is created, deleted, or changed.
+
+Create the **Notes viewmodel**:
+
+01. In the **Solution Explorer** pane of Visual Studio, double-click on **ViewModels\\NotesViewModel.cs**.
+01. Replace the code in this file with the following code:
+
+    ```csharp
+    using CommunityToolkit.Mvvm.Input;
+    using System.Collections.ObjectModel;
+    using System.Windows.Input;
+    
+    namespace Notes.ViewModels;
+    
+    internal class NotesViewModel: IQueryAttributable
+    {
+    }
+    ```
+
+    This code is the blank `NotesViewModel` where you'll add properties and commands to support the `AllNotes` view.
+
+01. In the `Notes` class code, add the following properties:
+
+    :::code language="csharp" source="../snippets/viewmodel-shared/csharp/Notes/ViewModels/NotesViewModel.cs" id="properties":::
+
+    The `AllNotes` property is an `ObservableCollection` that stores all of the notes loaded from the device. The two commands will be used by the view to trigger the actions of creating a note or selecting an existing note.
+
+01. Add a parameterless constructor to the class, which initializes the commands and loads the notes from the model:
+
+    :::code language="csharp" source="../snippets/viewmodel-shared/csharp/Notes/ViewModels/NotesViewModel.cs" id="ctor":::
+
+    Notice that the `AllNotes` collection uses the `Models.Note.LoadAll` method fill the observable collection with notes. The `LoadAll` method returns the notes as the `Models.Note` type, but the observable collection is a collection of `ViewModel.Note` types. The code uses the `Select` Linq extension to create viewmodel instances from the note models returned from `LoadAll`.
+
+01. Create the methods targeted by the commands:
+
+    :::code language="csharp" source="../snippets/viewmodel-shared/csharp/Notes/ViewModels/NotesViewModel.cs" id="commands":::
+
+    Notice that the `NewNoteAsync` method doesn't take a parameter while the `SelectNoteAsync` does. Commands can optionally have a single parameter that is provided when the command is invoked. For the `SelectNoteAsync` method, the parameter represents the note that's being selected.
+
+01. Finally, implement the `IQueryAttributable.ApplyQueryAttributes` method:
+
+    :::code language="csharp" source="../snippets/viewmodel-shared/csharp/Notes/ViewModels/NotesViewModel.cs" id="query":::
```

Comment:
Not sure why this is public?

Reply:
I finally see why NoteViewModel.Reload is public. But it looks like the only method that needs to be public?